lib/Optimizer/Transforms/QuakeSynthesizer.cpp

```diff
@@ -104,63 +104,58 @@ Value makeComplexElement(OpBuilder &builder, Location argLoc,
   return builder.create<complex::ConstantOp>(argLoc, eleTy, complexVal);
 }
 
-/// returns true if and only if \p argument is used by a `quake.init_state`
-/// operation.
-static bool hasInitStateUse(BlockArgument argument) {
```

Comment:
We definitely do not want all arrays to be converted to global constants.

The original purpose of this code still holds. Namely, we want to lift a vector of constants to a value so that all the constant propagation works correctly and we can build a proper quantum circuit.

Reply:
I'll add the check back.

Reply:
added the check back, so globals are only generated for args used in quake.init_state.